File: Aran/RNAV/Libs/Panda/Geometry/cpp/src/Ring.cpp
#include "../include/Ring.h"
#include "../include/GeometryType.h"

namespace Panda
{
	int Ring::geometryType () const throw ()
	{
		return GeometryType::Ring;
	}

	Geometry* Ring::clone () const throw (std::bad_cast)
	{
	    Ring* ring = new Ring;
        
        try
        {
    		ring->assign (*this);
        }
        catch (std::bad_cast e)
        {
            delete ring;
            throw e;
        }
        
		return ring;
	}
}

File: Aran/RNAV/Libs/Panda/Geometry/C#/Ring.cs
using System;
using System.Collections.Generic;
using System.Text;
using ARAN.Common;

namespace ARAN.GeometryClasses
{
	public class Ring : MultiPoint
	{
		public override Object Clone()
		{
			Ring rn;
			rn = new Ring();
			rn.Assign(this);
			return rn;
		}
		public override GeometryType GetGeometryType()
		{
			return GeometryType.Ring;
		}
	}
}
